mne/decoding/ssd.py

```diff
@@ -0,0 +1,320 @@
+       to the spectral ratio.
+       See Eq. (24) in :footcite:`NikulinEtAl2011`
+    return_filtered : bool (default True)
+        If return_filtered is True, data is bandpassed and projected onto
+        the SSD components.
+    n_fft: int (default None)
+       if sort_by_spectral_ratio is set to True, then the sources will be
+       sorted accordinly to their spectral ratio which is calculated based on
+       :func:`psd_array_welch` function. The n_fft parameter set the length of
+       FFT used. See :func:`mne.time_frequency.psd_array_welch` for more
+       information.
+    cov_method_params : dict | None (default None)
+        As in :func:`mne.decoding.SPoC`
+        The default is None.
+    rank : None | dict | ‘info’ | ‘full’
+        As in :func:`mne.decoding.SPoC`
+        This controls the rank computation that can be read from the
+        measurement info or estimated from the data.
+        See Notes of :func:`mne.compute_rank` for details.
+        We recomend to use 'full' when working with epoched data.
+
+    Attributes
+    ----------
+    filters_ : array, shape(n_channels, n_components)
+        The spatial filters to be multipled with the signal.
+    patterns_ : array, shape(n_components, n_channels)
+        The patterns for reconstructing the signal from the filtered data.
+
+    References
+    ----------
+    .. footbibliography::
+    """
+
+    def __init__(self, info, filt_params_signal, filt_params_noise,
+                 estimator='oas', n_components=None, picks=None,
+                 sort_by_spectral_ratio=True, return_filtered=False,
+                 n_fft=None, cov_method_params=None, rank=None):
+        """Initialize instance"""
+
+        dicts = {"signal": filt_params_signal, "noise": filt_params_noise}
+        for param, dd in [('l', 0), ('h', 0), ('l', 1), ('h', 1)]:
+            key = ('signal', 'noise')[dd]
+            if param + '_freq' not in dicts[key]:
+                raise ValueError(
+                    "%s must be defined in filter parameters for %s"
+                    % (param + '_freq', key))
+            val = dicts[key][param + '_freq']
+            if not isinstance(val, (int, float)):
+                raise ValueError(
+                    "Frequencies must be numbers, got %s" % type(val))
+        # check freq bands
+        if (filt_params_noise['l_freq'] > filt_params_signal['l_freq'] or
+                filt_params_signal['h_freq'] > filt_params_noise['h_freq']):
+            raise ValueError('Wrongly specified frequency bands!\n'
+                    'The signal band-pass must be within the noise band-pass!')
+        ch_types = {channel_type(info, ii)
+                    for ii in range(info['nchan'])}
+        if len(ch_types) > 1:
+            raise ValueError("At this point SSD only supports fitting "
+                             "single channel types. Your info has %i types" %
+                             (len(ch_types)))
```

Comment:
We never test that these Errors work as expected, right? Something we should add. Provoke, error, catch, check if  message is as expected. See other API tests around for inspiration.